### PR TITLE
Eliminate the loop in SyncRes::getAddrs()

### PR DESCRIPTION
**Pull request by omoerbeek**
Makes the code much more readable.

### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [ ] included documentation (including possible behaviour changes)
- [X] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/syncres.cc [hunk 1]
```diff
@@ -680,61 +680,54 @@ struct speedOrderCA
...
-	      }
-	    }
-	  }
-	}
+  }
+  if (s_doIPv6) {
+    if (ret.empty()) {
+      // We did not find IPv4 addresses, try to get IPv6 ones
+      vState newState = Indeterminate;
+      res_t res;
```
Comment (rgacogne) [inline pdns/syncres.cc]: The two declarations above shadow existing variables, we should avoid that whenever possible.

File: pdns/syncres.cc [hunk 2]
```diff
@@ -680,61 +680,54 @@ struct speedOrderCA
...
-	  if(auto rec = getRR<ARecordContent>(*i))
-	    ret.push_back(rec->getCA(53));
-	  else if(auto aaaarec = getRR<AAAARecordContent>(*i))
-	    ret.push_back(aaaarec->getCA(53));
-          done=true;
+  vState newState = Indeterminate;
+  res_t res;
+  // If IPv4 ever becomes second class, we should revisit this
+  if (doResolve(qname, QType::A, res, depth+1, beenthere, newState) == 0) {  // this consults cache, OR goes out
+    for (res_t::const_iterator i = res.begin(); i != res.end(); ++i) {
```
Comment (rgacogne) [inline pdns/syncres.cc]: Just a nit but it looks like a C++11-style loop would be easier to read.

File: pdns/syncres.cc (hunk 1 above)
Comment (omoerbeek) [reply pdns/syncres.cc]: Agreed, will fix.

File: pdns/syncres.cc (hunk 2 above)
Comment (omoerbeek) [reply pdns/syncres.cc]: Right, will do.

File: pdns/syncres.cc [hunk 3]
```diff
@@ -680,61 +680,54 @@ struct speedOrderCA
...
-	  if(auto rec = getRR<ARecordContent>(*i))
-	    ret.push_back(rec->getCA(53));
-	  else if(auto aaaarec = getRR<AAAARecordContent>(*i))
-	    ret.push_back(aaaarec->getCA(53));
-          done=true;
+  vState newState = Indeterminate;
+  res_t resv4;
+  // If IPv4 ever becomes second class, we should revisit this
+  if (doResolve(qname, QType::A, resv4, depth+1, beenthere, newState) == 0) {  // this consults cache, OR goes out
+    for (auto i = resv4.cbegin(); i != resv4.end(); ++i) {
```
Comment (rgacogne) [inline pdns/syncres.cc]:
```suggestion
    for (const auto& i : resv4) {
```

Note that `i` becomes a const reference to the actual type, and not an iterator anymore so it implies editing a bit more code than just that line.

**Pull request merged**